testutil/accumulator.go

```diff
@@ -161,6 +161,17 @@ func (a *Accumulator) Get(measurement string) (*Metric, bool) {
 	return nil, false
 }
 
+// Calls the given Gather function and returns the first error found.
+func (a *Accumulator) GatherError(gf func(telegraf.Accumulator) error) error {
+	if err := gf(a); err != nil {
+		return err
+	}
+	if len(a.Errors) > 0 {
+		return a.Errors[0]
+	}
+	return nil
+}
```

Comment:
Another improvement might be to add an `acc.AddErrorf()`. The vast majority of the usage is nesting an `fmt.Errorf()` inside `acc.AddError()`. Would just be a convenience.